java/spark/v3/src/main/java/com/lancedb/lance/spark/LanceColumnarPartitionReader.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.lancedb.lance.spark;
+
+import com.lancedb.lance.spark.internal.LanceFragmentColumnarBatchScanner;
+import org.apache.spark.sql.connector.read.PartitionReader;
+import org.apache.spark.sql.vectorized.ColumnarBatch;
+
+import java.io.IOException;
+
+public class LanceColumnarPartitionReader implements PartitionReader<ColumnarBatch> {
+  private final LanceInputPartition inputPartition;
+  private int fragmentIndex;
+  private LanceFragmentColumnarBatchScanner fragmentReader;
+  private ColumnarBatch currentBatch;
+
+  public LanceColumnarPartitionReader(LanceInputPartition inputPartition) {
+    this.inputPartition = inputPartition;
+    this.fragmentIndex = 0;
+  }
+
+  @Override
+  public boolean next() throws IOException {
+    if (fragmentReader != null && fragmentReader.loadNextBatch()) {
+      if (currentBatch != null) {
```

Comment:
can we just call `close()` instead, instead of 
`if (currentBatch != null) {
          currentBatch.close();
        }`